transformers/uws/server.js

```diff
@@ -46,7 +46,12 @@ module.exports = function server() {
       });
     });
 
-    this.service.upgrade(soc._handle.fd, req.headers['sec-websocket-key'], soc.ssl ? soc.ssl._external : null);
+    var ticket = this.service.transfer(soc._handle.fd, soc.ssl ? soc.ssl._external : null);
+    soc.on('close', function(error) {
+      if (!error) {
```

Comment:
We are definitely the first to call [`socket.destroy()`](https://github.com/nodejs/node/blob/a2e57192ebf79084f34766a89e8082067f48af14/lib/net.js#L516-L519) and to set the socket as [`destroyed`](https://github.com/nodejs/node/blob/a2e57192ebf79084f34766a89e8082067f48af14/lib/net.js#L502).
The  [`close`](https://github.com/nodejs/node/blob/a2e57192ebf79084f34766a89e8082067f48af14/lib/net.js#L492) event is only emitted in the `_destroy` method and any other call after the one we made will find the socket already [`destroyed`](https://github.com/nodejs/node/blob/a2e57192ebf79084f34766a89e8082067f48af14/lib/net.js#L469-L473) preventing any other `close` event from being emitted.

I think there is no way we can receive a `close` event with a `true` argument.

Actually it doesn't event matter, it is not something that will affect performance, it only caught my attention because it seems redundant.